### DISCUSSION TITLE
LDAP auth?

### DISCUSSION THREAD

**Discussion by sendtoworthington** [Lissy93/dashy#372] [Ideas]
How can we implement use of LDAP for user authentication? I already had LDAP set up for other apps and I don't want to spin up a Keycloak app just for user auth. Any thoughts if this could be an official option?

**Comment by lissy93** [comment]
I would be happy to review a pull request, but this isn't currently something that I've seen many other requests for, so probably won't be implementing it myself in the near future. Especially since LDAP and Active Directory are actually [supported by KeyCloack](https://wjw465150.gitbooks.io/keycloak-documentation/content/server_admin/topics/user-federation/ldap.html), which in turn is supported by Dashy.

Sorry about that. But if you need any support with adding it, I would be happy to help :)

I'd suggest looking into [Node-LDAP](https://github.com/ldapjs/node-ldapjs), which uses the [OpenLDAP](https://www.openldap.org/) server. This can be implemented in [`server.js`](https://github.com/Lissy93/dashy/blob/master/server.js).
